### PR TITLE
Transitioned the register tables of non topgen blocks to CMDGEN

### PR DESCRIPTION
**Pull request by HU90m**
Continuation of https://github.com/lowRISC/opentitan/pull/18694 and https://github.com/lowRISC/opentitan/pull/19266 .

This will fix https://github.com/lowRISC/opentitan/issues/19183 for the transitioned blocks.

The only blocks remaining are `ip_template`s (`alert_handler` and `rv_plic`). I'll likely have to make a small change to the `topgen` tool to transition these blocks.

Had to edit hjson files to remove trailing whitespace.

### CODE REVIEW

File: hw/ip/spi_host/data/spi_host.hjson [hunk 1]
```diff
@@ -212,8 +210,7 @@
         { bits: "24",
           name: "RXEMPTY",
           desc: '''When high, indicates that the receive fifo is empty.
-                   Any reads from RX FIFO will cause an error interrupt.
-                   '''
+                   Any reads from RX FIFO will cause an error interrupt.'''
```
Comment (jwnrt) [inline hw/ip/spi_host/data/spi_host.hjson:213]: What was the motivation for these changes?

The [Hjson spec](https://hjson.github.io/syntax.html#multiline-strings) is a little hard to understand on multiline strings, but I think formatting like this is safest:

```
desc:
  '''
  When high, indicates that the receive fifo is empty.
  Any reads from RX FIFO will cause an error interrupt.
  '''
```

Regardless, if this works then go ahead with it - we can get something more stable in the future with a Hjson linter.

**Review by jwnrt — APPROVED** [review]
LGTM assuming the link check passes. Thanks @HU90m.

Is this the last of these changes?

File: hw/ip/spi_host/data/spi_host.hjson (hunk 1 above)
Comment (jwnrt) [reply hw/ip/spi_host/data/spi_host.hjson:213]: Also, it seems we're on RTL lockdown so make sure to tag someone who can override the block or defer the Hjson changes

Comment (HU90m) [reply hw/ip/spi_host/data/spi_host.hjson:213]: I tried to make the minimal changes required to remove the whitespaces. I agree, the triple quotes on their own line is nicer.

**Comment by HU90m** [comment]
> Is this the last of these changes?

For this PR, yes.

> Anything else I should have checked?

Nope, that all sounds brilliant. Thank you!

File: hw/ip/spi_device/README.md [hunk 2]
```diff
@@ -76,7 +76,7 @@ The IP supports Normal Read, Fast Read, Fast Read Dual Output, Fast Read Quad Ou
 This version of IP does not support Dual IO, Quad IO, QPI commands.
 
 In Passthrough mode, SPI Device receives SPI transactions from a host system and forwards the transactions to a downstream flash device.
-SW may filter prohibited commands by configuring 256-bit [`FILTER`](data/spi_device.hjson#filter) CSR.
+SW may filter prohibited commands by configuring 256-bit [`FILTER`](doc/registers.md#filter) CSR.
```
Comment (hcallahan-lowrisc) [inline hw/ip/spi_device/README.md:79]: This anchor does not exist. How about...
```suggestion
SW may filter prohibited commands by configuring the 256-bit FILTER CSR ([`CMD_FILTER_0`](doc/registers.md#cmd_filter_0) through ['CMD_FILTER_7'](doc/registers.md#cmd_filter_7)).
```

File: hw/ip/spi_device/doc/registers.md [hunk 3]
```diff
@@ -0,0 +1,1791 @@
...
+| spi_device.[`JEDEC_ID`](#jedec_id)                       | 0x44     |        4 | JEDEC ID register.                                      |
+| spi_device.[`READ_THRESHOLD`](#read_threshold)           | 0x48     |        4 | Read Buffer threshold register.                         |
+| spi_device.[`MAILBOX_ADDR`](#mailbox_addr)               | 0x4c     |        4 | Mailbox Base address register.                          |
+| spi_device.[`UPLOAD_STATUS`](#upload_status)             | 0x50     |        4 | Upload module status register.                          |
+| spi_device.[`UPLOAD_STATUS2`](#upload_status2)           | 0x54     |        4 | Upload module status 2 register.                        |
+| spi_device.[`UPLOAD_CMDFIFO`](#upload_cmdfifo)           | 0x58     |        4 | Command Fifo Read Port.                                 |
+| spi_device.[`UPLOAD_ADDRFIFO`](#upload_addrfifo)         | 0x5c     |        4 | Address Fifo Read Port.                                 |
+| spi_device.[`CMD_FILTER_0`](#CMD_FILTER_0)               | 0x60     |        4 | Command Filter                                          |
```
Comment (hcallahan-lowrisc) [inline hw/ip/spi_device/doc/registers.md:32]: The fragments here need to be lowercase, as the url fragments are case-sensitive.

File: hw/ip/spi_device/README.md (hunk 2 above)
Comment (HU90m) [reply hw/ip/spi_device/README.md:79]: For this PR, I've avoided making any non-trivial register link fixes. There are a few registers linked to that no longer exist. Fragment checking in CI is in the pipeline.

**Comment by HU90m** [comment]
@mundaym please could you authorize the changes to the hjson files. There is no function difference, just whitespace fixes.

**Comment by mundaym** [comment]
CHANGE AUTHORIZED: hw/ip/hmac/data/hmac.hjson
CHANGE AUTHORIZED: hw/ip/pwm/data/pwm.hjson
CHANGE AUTHORIZED: hw/ip/rv_core_ibex/data/rv_core_ibex.hjson
CHANGE AUTHORIZED: hw/ip/spi_host/data/spi_host.hjson
CHANGE AUTHORIZED: hw/ip/sysrst_ctrl/data/sysrst_ctrl.hjson

**Comment by mundaym** [comment]
CHANGE AUTHORIZED: hw/ip/hmac/data/hmac.hjson
CHANGE AUTHORIZED: hw/ip/pwm/data/pwm.hjson
CHANGE AUTHORIZED: hw/ip/rv_core_ibex/data/rv_core_ibex.hjson
CHANGE AUTHORIZED: hw/ip/spi_host/data/spi_host.hjson
CHANGE AUTHORIZED: hw/ip/sysrst_ctrl/data/sysrst_ctrl.hjson

**Comment by msfschaffner** [comment]
CHANGE AUTHORIZED: hw/ip/hmac/data/hmac.hjson
CHANGE AUTHORIZED: hw/ip/pwm/data/pwm.hjson
CHANGE AUTHORIZED: hw/ip/rv_core_ibex/data/rv_core_ibex.hjson
CHANGE AUTHORIZED: hw/ip/spi_host/data/spi_host.hjson
CHANGE AUTHORIZED: hw/ip/sysrst_ctrl/data/sysrst_ctrl.hjson

**Review by msfschaffner — APPROVED** [review]

**Comment by msfschaffner** [comment]
Thanks for the cleanup, @HU90m!

**Pull request merged**